draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
 is resource-dependent.  In the absense of any resource-specific requirements,
-any field not understood by implementations MUST be ignored.
+any member not understood by implementations MUST be ignored.
+
+[[links]]
+== Links
+
+An Ion Link is an Ion Object where its `meta` member contains an {uri-IRIs}[IRI] member named `href`. An Ion Link is therefore JSON serialization of a {uri-rfc5988}[web link] to another resource.
+
+Ion parsers MUST identify any JSON structure matching this definition as an Ion Link.  Anything that does not match this defined structure MUST NOT be considered an Ion Link.
+
+A non-normative example of a minimal Ion link:
+
+.Example simplest Ion link:
+[source,json]
+----
+{ "meta": { "href": "https://api.example.com/corporations/acme" } }
+----
+
+=== Link Relation Types
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
+
+===== Collection Item Implicit Link Relation Type
+
+Any Ion Link in a Collection Resource's `items` array is implicitly assigned the {uri-linkRelType-item}[`item`] link relation type.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users" },
+  "items": [
+    { "meta": { "href": "https://api.example.com/users/1" } },
+    { "meta": { "href": "https://api.example.com/users/2" } }
+  ]
+}
+----
+
+The above example has 2 Ion Links in its `items` array.  Each of these 2 links are implicitly assigned the {uri-linkRelType-item}[`item`] link relation type.
```

Comment:
I had intended that the 'auto assign the link relation type' would be limited specifically to the member attribute named `items` only and only if that `items` member was part of a _Collection Resource_.  This avoids any other interpretation and keeps the scope focused.

I'm a fan of not opening up to other edge cases unless there is a strong reason to do so.

As you point out, there can be a conflict otherwise, so I don't see much value in opening  at the moment.  Thoughts?  I definitely can be persuaded otherwise if there are good use cases.